src/Controls/src/Core/Button/Button.iOS.cs

```diff
@@ -464,6 +464,9 @@ private static void MapPadding(IButtonHandler handler, Button button)
 		public static void MapText(IButtonHandler handler, Button button)
 		{
 			handler.PlatformView?.UpdateText(button);
+
+			// Any text update requires that we update any attributed string formatting
+			handler.PlatformView?.UpdateCharacterSpacing(button);
```

Comment:
I think this happen here https://github.com/dotnet/maui/blob/a11555b9c8dfd7a80f89917eab7b9558c442bfff/src/Controls/src/Core/Platform/iOS/Extensions/ButtonExtensions.cs#L98 before update ContentLayout. The ContentLayout calculations are based on the text Frame and the character spacing have an impact on it.